src/main/scala/tilelink/Parameters.scala

```diff
@@ -1087,6 +1199,64 @@ case class TLEdgeParameters(
+  def supportsLogicalSafe   (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(true,  _.supports.logical,    address, lgSize, range) && emitHelper(true, _.emits.logical,    address, lgSize, range)
+  def supportsGetSafe       (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(true,  _.supports.get,        address, lgSize, range) && emitHelper(true, _.emits.get,        address, lgSize, range)
+  def supportsPutFullSafe   (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(true,  _.supports.putFull,    address, lgSize, range) && emitHelper(true, _.emits.putFull,    address, lgSize, range)
+  def supportsPutPartialSafe(address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(true,  _.supports.putPartial, address, lgSize, range) && emitHelper(true, _.emits.putPartial, address, lgSize, range)
+  def supportsHintSafe      (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(true,  _.supports.hint,       address, lgSize, range) && emitHelper(true, _.emits.hint,       address, lgSize, range)
+
+  def supportsAcquireTFast  (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(false, _.supports.acquireT,   address, lgSize, range) && emitHelper(false, _.emits.acquireT,   address, lgSize, range)
+  def supportsAcquireBFast  (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(false, _.supports.acquireB,   address, lgSize, range) && emitHelper(false, _.emits.acquireB,   address, lgSize, range)
+  def supportsArithmeticFast(address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(false, _.supports.arithmetic, address, lgSize, range) && emitHelper(false, _.emits.arithmetic, address, lgSize, range)
+  def supportsLogicalFast   (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(false, _.supports.logical,    address, lgSize, range) && emitHelper(false, _.emits.logical,    address, lgSize, range)
+  def supportsGetFast       (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(false, _.supports.get,        address, lgSize, range) && emitHelper(false, _.emits.get,        address, lgSize, range)
+  def supportsPutFullFast   (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(false, _.supports.putFull,    address, lgSize, range) && emitHelper(false, _.emits.putFull,    address, lgSize, range)
+  def supportsPutPartialFast(address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(false, _.supports.putPartial, address, lgSize, range) && emitHelper(false, _.emits.putPartial, address, lgSize, range)
+  def supportsHintFast      (address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = supportHelper(false, _.supports.hint,       address, lgSize, range) && emitHelper(false, _.emits.hint,       address, lgSize, range)
```

Comment:
I think these should be called something else, that suggests it requires both supports and emits. Maybe expectsGetFast? Or something else @hcook ?

Reply:
i like expects